packages/mui-docs/src/MarkdownElement/MarkdownElement.tsx

```diff
@@ -823,21 +825,17 @@ export interface MarkdownElementProps {
 export const MarkdownElement = React.forwardRef<HTMLDivElement, MarkdownElementProps>(
   function MarkdownElement(props, ref) {
     const { className, renderedMarkdown, ...other } = props;
+    const rootRef = React.useRef<HTMLElement>(null);
+    const handleRef = useForkRef(rootRef, ref);
 
     React.useEffect(() => {
-      const elements = document.getElementsByClassName('title-link-to-anchor');
+      const elements = rootRef.current!.getElementsByClassName('title-link-to-anchor');
```

Comment:
The CSS is scoped to the component, so much better not to leak outside of this scope, esptially since we bypass React here.